dts/bindings/misc/zephyr,flash-disk.yaml

```diff
@@ -20,7 +20,7 @@ properties:
       description: |
         Disk name.
 
-    sector-size:
+    disk-sector-size:
```

Comment:
This is property of a disk, it is obvious, therefore a "disk" prefix here is not necessary. Also, see description below.
Note also that more unique "block-size" was not found good enough.

Reply:
While I agree that it is a disk property the disk-prefix should not be required it is added because it might be confusing with the backend that is used that also has a sector-size (at least for spi-nor-flash).
The reason sector-size is used is because this is what a disk reports, it doesn't report block-size, and the code is using the sector-size at multiple locations.

Reply:
I agree with @jfischer-no . we already agree what does `sector-size` or `page-size`, although I still don't like it, because of flash documentation and JEDEC standards, the last one actually clarifies what the sector means (JESD260E, page 53, paragraph 6.5).

In case of disk definition `sector-size` meaning is provided within context of `zephyr,flash-disk`:
https://github.com/zephyrproject-rtos/zephyr/blob/51d510adc60f6bb70c58384328e3661e86d0b13c/dts/bindings/misc/zephyr%2Cflash-disk.yaml#L23-L27.

The "fat-standard-sector" is logical sector size, and is actually software override of hardware characteristic, that bundles several hardware sectors together; do not confuse with cluster which is another thing.

Reply:
I disagree, in the binding we are defining the disk sector-size as it is used by the IO_STAT interface, it is not directly related to any JEDEC standard. However the disk sector-size is defined in the context of a flash backend that also uses the terminology of sector-size (as defined in the JEDEC standard), therefore it is wise to make the description more clear.

Reply:
What about going with `emulated-sector-size`? Then it shouldn't be confused with any other `sector-size` and the emulation context seems clear.